### PR TITLE
feat(core): remove debug decorator

### PR DESCRIPTION
**Pull request by tohrnii**
Remove debug decorator as it is no longer required since `execute_iter` gives all the same debug related info.
Partly addresses #137.

### CODE REVIEW

File: core/src/program/blocks/span_block.rs [hunk 1]
```diff
@@ -659,25 +658,6 @@ mod tests {
         assert_eq!(hasher::hash_elements(&all_groups[..10]), hash);
     }
 
-    #[test]
-    fn batch_ops_with_decorator() {
-        let ops = vec![
-            Operation::Push(Felt::ONE),
-            Operation::Add,
-            Operation::Debug(DebugOptions::All),
```
Comment (bobbinth) [inline core/src/program/blocks/span_block.rs:667]: I'd prefer not to remove this test. Maybe just replace the `Debug` decorator with `Advice` decorator?

**Review by bobbinth — CHANGES_REQUESTED** [review]
Looks good! One small comment inline. Also, could you update [mdBook docs](https://maticnetwork.github.io/miden/user_docs/assembly/debugging.html) to remove the section about debugging?

File: core/src/program/blocks/span_block.rs (hunk 1 above)
Comment (tohrnii) [reply core/src/program/blocks/span_block.rs:667]: Fixed.

**Review by bobbinth — APPROVED** [review]
Looks good! Thank you! Could you squash the last couple of commits? I'll merge after that.

**Pull request merged**